libs/events/src/CowEventEmitter.ts

```diff
@@ -0,0 +1,47 @@
+import { CowEvents, CowEventPayloads } from './types'
+
+export type CowEventHandler<T extends keyof CowEventPayloads> = (payload: CowEventPayloads[T]) => void
+
+export type CowEventListener<T extends CowEvents> = T extends keyof CowEventPayloads
+  ? { event: T; handler: CowEventHandler<T> }
+  : never
+
+export type CowEventListeners = CowEventListener<CowEvents>[]
+
+export interface CowEventEmitter {
+  on<T extends CowEvents>(event: T, handler: CowEventHandler<T>): void
+  emit<T extends CowEvents>(event: T, payload: CowEventPayloads[T]): void
+  off<T extends CowEvents>(event: T, listenerToRemove: CowEventHandler<T>): void
+}
+
+export class CowEventEmitterImpl {
```

Comment:
Shouldn't this implementation implement the interface?

```suggestion
export class CowEventEmitterImpl implements CowEventEmitter {
```

Reply:
Yes, I saw it later. Recent PRs do it